runtime/sema/checker.go

```diff
@@ -241,7 +242,17 @@ func WithPositionInfoEnabled(enabled bool) Option {
 //
 func WithLintingEnabled(enabled bool) Option {
 	return func(checker *Checker) error {
-		checker.lintEnabled = enabled
+		checker.lintingEnabled = enabled
+		return nil
+	}
+}
+
+// WithErrorShortCircuitingEnabled returns a checker option which enables/disables
+// error short-circuiting.
```

Comment:
The function name does a pretty good job of communicating this. I recommend describing error short-circuiting:
```suggestion
// WithErrorShortCircuitingEnabled returns a checker option which enables/disables
// error short-circuiting in the checker.
// When enabled, the checker will stop running once it encounters an error.
// When disabled (the default), the checker reports the error then continues checking.
```

Reply:
Nice! Adding